client/src/pages/Challenge.tsx

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+import { Container, Row, Col, ButtonToolbar, Button } from "react-bootstrap";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { ChallengeDetails, ChallengeDifficulties } from "../types";
+import { getDifficulty } from "../utils";
+
+const Challenge = () => {
+
+    const [completed, setCompleted] = useState(true);
+    const [details, setDetails]  = useState<ChallengeDetails>();
+    const [showingDetails, setShowingDetails] = useState(false); //toggle state for showing/hiding design pattern list
+
+    const { id } = useParams();
+
+    useEffect(() => {   
+        //fetch data about the challenge with the provided "id"
+        //NEEDS TESTING
+        //get the challenge details
+        fetch('/api/challenge/' + id).then((response) => {
+            if(!response.ok){
+                throw new Error(response.statusText);                
+            }
+            return response.json() as Promise<{data : ChallengeDetails}>; //THIS LINE MAY CAUSE ERRORS. Need to test with proper server.
+        }).then((data) => {
+            setDetails(data.data);
+            return;
+        }).catch((err) => {
+            console.log();
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            console.error("Failed fetching the challenge number: " + id + "\nError message: " + err.message)
+        });
+   }, [id])
+
+    //START FOR TESTING ONLY
+
+    const challengeDetails: ChallengeDetails = {
```

Comment:
Extract the testing default into a const outside and set it as the default state value. Keeping it inside the components makes it harder to read.